app/auth/middleware.js

```diff
@@ -11,23 +11,25 @@ function processAuthResponse() {
     }
 
     try {
-      const userLocations = await userLocationService.getUserLocations(
-        grant.response.access_token
-      )
+      const accessToken = grant.response.access_token
+      const decodedAccessToken = decodeAccessToken(grant.response.access_token)
+      const [locations, fullName] = await Promise.all([
+        getLocations(accessToken),
+        getFullName(accessToken),
+      ])
 
       req.session.regenerate(error => {
         if (error) {
           return next(error)
         }
 
-        const accessToken = decodeAccessToken(grant.response.access_token)
-
-        req.session.authExpiry = accessToken.exp
+        req.session.authExpiry = decodedAccessToken.exp
         req.session.postAuthRedirect = postAuthRedirect
         req.session.user = new User({
-          name: accessToken.user_name,
-          roles: accessToken.authorities,
-          locations: userLocations,
+          fullName,
+          locations,
+          userName: decodedAccessToken.user_name,
```

Comment:
I know it's not fully related to this change but `user.userName` should probably be all lowercase as it's one word, not two.